profiles/models/scope.py

```diff
@@ -0,0 +1,114 @@
+from django.db.models import CharField, Model, ManyToManyField, Prefetch, Q, ForeignKey
+from django.contrib.auth.models import User
+from django.urls import reverse
+
+from Squest.utils.squest_model import SquestModel
+from profiles.models import Role
+from profiles.models.rbac import RBAC
+
+
+class AbstractScope(SquestModel):
+    name = CharField(max_length=500)
+    description = CharField(max_length=500, blank=True)
+    roles = ManyToManyField(
+        Role,
+        blank=True,
+        help_text="The roles assign to the scope.",
+        related_name="scopes",
+        related_query_name="scopes",
+        verbose_name="Default roles"
+    )
+
+    def get_object(self):
+        if hasattr(self, "scope"):
+            return self.scope.get_object()
+        elif hasattr(self, "globalpermission"):
+            return self.globalpermission
+        raise Exception("This scope is not implemented")
+
+    def get_queryset_for_user(cls, user, perm):
+        qs = super().get_queryset_for_user(user, perm)
+        if qs.exists():
+            return qs
+        app_label, codename = perm.split(".")
+        return Scope.objects.filter(rbac__user=user,
+                                    rbac__role__permissions__codename=codename,
+                                    rbac__role__permissions__content_type__app_label=app_label) | \
+               Scope.objects.filter(Q(team__org__rbac__user=user),
+                                    scope__rbac__role__permissions__codename=codename,
+                                    scope__rbac__role__permissions__content_type__app_label=app_label)
+
+    def get_scopes(self):
+        return self.get_object().get_scopes()
+
+    def __str__(self):
+        return str(self.get_object())
+
+    @property
+    def users(self):
```

Comment:
add a comment that explain the research we are doing here.
Like: We get all users that are in a group which has at least one permission on the target scope